modules/reactor/src/meshgenerators/CoreMeshGenerator.C

```diff
@@ -354,81 +391,118 @@ std::unique_ptr<MeshBase>
 CoreMeshGenerator::generate()
 {
   // This generate() method will be called once the subgenerators that we depend on are
-  // called. This is where we set the region ids and element integers.
-
+  // called. This is where we reassign subdomain ids/names in case they were merged
+  // when stitching assemblies into the core. This is also where we set region_id extra
+  // element integers, which has not been set yet for extruded geometries
+
+  // Define all extra element names and integers
+  std::string pin_type_id_name = "pin_type_id";
+  std::string assembly_type_id_name = "assembly_type_id";
+  std::string plane_id_name = "plane_id";
+  std::string region_id_name = "region_id";
+  std::string radial_id_name = "radial_id";
+  const std::string default_block_name = "RGMB_CORE";
+
+  auto pin_type_id_int = getElemIntegerFromMesh(*(*_build_mesh), pin_type_id_name, true);
+  auto assembly_type_id_int = getElemIntegerFromMesh(*(*_build_mesh), assembly_type_id_name, true);
+  auto radial_id_int = getElemIntegerFromMesh(*(*_build_mesh), radial_id_name, true);
+  auto region_id_int = getElemIntegerFromMesh(*(*_build_mesh), region_id_name, true);
+  unsigned int plane_id_int = 0;
   if (_extrude)
+    plane_id_int = getElemIntegerFromMesh(*(*_build_mesh), plane_id_name, true);
+
+  // Get next free block ID in mesh in case subdomain ids need to be remapped
+  auto next_block_id = MooseMeshUtils::getNextFreeSubdomainID(*(*(_build_mesh)));
+  std::map<std::string, SubdomainID> rgmb_name_id_map;
+
+  // Loop through all mesh elements and set region ids and reassign block IDs/names
+  // if they were merged during assembly stitching
+  for (auto & elem : (*_build_mesh)->active_element_ptr_range())
```

Comment:
this could be real expensive in 3D after the extrusion.

Fixing the merging of names could be done in 2D right? What cannot be done in 2D?

Reply:
Well both the region ID and the block names depends on the user input in the pin MG, which are 2D vector axial * radial.

Save for writing a new extruder that also takes care of adding those, not sure what you can do about this?

Reply:
Yeah, right now this step is needed to define region IDs and block names for all extruded elements